src/main/java/emissary/kff/KffFile.java

```diff
@@ -143,42 +138,34 @@ public String getPreferredAlgorithm() {
      * @param crc Result of CRC calculation
      * @return true if the record is in the list, false if it isn't
      */
-    private boolean binaryFileSearch(@Nullable byte[] hash, long crc) {
-        if (hash == null && myPreferredAlgorithm != null) {
-            logger.error("Unable to get digest computation for " + myPreferredAlgorithm);
-            return false;
-        }
+    private boolean binaryFileSearch(@Nonnull byte[] hash, long crc) {
 
         // Initialize indexes for binary search
         int low = 0;
         int high = bSearchInitHigh;
 
         /* Buffer to hold a record */
-        byte[] record = new byte[recordLength];
```

Comment:
`record` is a restricted identifier and should not be used as an identifier (https://rules.sonarsource.com/java/RSPEC-6213)